src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -136,6 +138,36 @@ internal SourceText ReadFileContent(CommandLineSourceFile file, IList<Diagnostic
             return ReadFileContent(file, diagnostics, out discarded);
         }
 
+        internal EmbeddedText ReadEmbeddedFileContent(CommandLineSourceFile file, IList<DiagnosticInfo> diagnostics)
```

Comment:
>ReadEmbeddedFileContent [](start = 30, length = 23)

Perhaps rename to TryReadEmbeddedFileContent to emphasize that it may fail and return null.